tests/integration/jit/function/test_span_events_are_created_upon_caught_error.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*DESCRIPTION
+Test that span events are correctly created from any eligible segment, even
+when an error is generated and handled.
+*/
+
+/*SKIPIF
+<?php
+
+require('skipif.inc');
+
+*/
+
+/*INI
+newrelic.distributed_tracing_enabled=1
+newrelic.transaction_tracer.threshold = 0
+newrelic.span_events_enabled=1
+newrelic.cross_application_tracer.enabled = false
+display_errors=1
+log_errors=0
+error_reporting = E_ALL
+opcache.enable=1
+opcache.enable_cli=1
+opcache.file_update_protection=0
+opcache.jit_buffer_size=32M
+opcache.jit=function
+*/
+
+/*PHPMODULES
+zend_extension=opcache.so
+*/
+
+/*EXPECT_SPAN_EVENTS
+[
+  "?? agent run id",
+  {
+    "reservoir_size": 10000,
+    "events_seen": 4
+  },
+  [
+    [
+      {
+        "traceId": "??",
+        "duration": "??",
+        "transactionId": "??",
+        "name": "OtherTransaction\/php__FILE__",
+        "guid": "??",
+        "type": "Span",
+        "category": "generic",
+        "priority": "??",
+        "sampled": true,
+        "nr.entryPoint": true,
+        "timestamp": "??",
+        "transaction.name": "OtherTransaction\/php__FILE__"
+      },
+      {},
+      {}
+    ],
+    [
+      {
+        "type": "Span",
+        "traceId": "??",
+        "transactionId": "??",
+        "sampled": true,
+        "priority": "??",
+        "name": "Datastore\/statement\/FakeDB\/other\/other",
+        "guid": "??",
+        "timestamp": "??",
+        "duration": "??",
+        "category": "datastore",
+        "parentId": "??",
+        "span.kind": "client",
+        "component": "FakeDB"
+      },
+      {},
+      {
+        "db.instance": "unknown",
+        "peer.hostname": "unknown",
+        "peer.address": "unknown:unknown"
+      }
+    ],
+    [
+      {
+        "type": "Span",
+        "traceId": "??",
+        "transactionId": "??",
+        "sampled": true,
+        "priority": "??",
+        "name": "Custom\/a",
+        "guid": "??",
+        "timestamp": "??",
+        "duration": "??",
+        "category": "generic",
+        "parentId": "??"
+      },
+      {},
+      {
+        "error.message": "foo",
+        "error.class": "E_USER_ERROR",
+        "code.lineno": "??",
+        "code.filepath": "__FILE__",
+        "code.function": "??"
+      }
+    ],
+    [
+      {
+        "type": "Span",
+        "traceId": "??",
+        "transactionId": "??",
+        "sampled": true,
+        "priority": "??",
+        "name": "Custom\/{closure}",
+        "guid": "??",
+        "timestamp": "??",
+        "duration": "??",
+        "category": "generic",
+        "parentId": "??"
+      },
+      {},
+      {
+        "code.lineno": "??",
+        "code.filepath": "__FILE__",
+        "code.function": "??"
+      }
+    ]
+  ]
+]
+*/
+
+/*EXPECT_REGEX
+^\s*(PHP )?Fatal error:\s*foo in .*? on line [0-9]+\s*$
+*/
+
+set_error_handler(
+    function () {
```

Comment:
Technically, since this PR fixes the invalid code in tests that set user exception handler, I think it makes sense to fix error handler callback signature as well. According to https://www.php.net/manual/en/function.set-error-handler.php, the callback should accept at least two parameters: `int $errno` and `string $errstr` (the others - `string $errfile`, `int $errline`, `array $errcontext` - are indicated as optional). Therefore consider this change:
```suggestion
    function (int $errno, string $errstr) {
```